### ISSUE TITLE
support loading css first

### ISSUE THREAD

**Issue by dominikg** [sveltejs/vite-plugin-svelte#1032]
### Describe the problem

Currently css is cached and returned as a virtual module when `.svelte` files are imported as js module.
But with vite `?direct` or `?inline` query urls, it is possible that the first request is not for the js module but the css.

### Describe the proposed solution

In case the cache is empty when css is requested, compile.

### Alternatives considered

keep current behavior, styles might not be needed before the js module

### Importance

nice to have

**Comment by dominikg** [comment]
see #341 #1024

**Comment by dominikg** [comment]
.svelte files never end up in the browser cache. do you mean js modules compiled from .svelte files? please file a separate issue with a reproduction for this.

**Comment by SystemParadox** [comment]
> .svelte files never end up in the browser cache. do you mean js modules compiled from .svelte files? please file a separate issue with a reproduction for this.

Yes. I've opened https://github.com/vitejs/vite/issues/19632 because I think it's a general vite issue. But it would have helped a lot with debugging this if vite-plugin-svelte had given a clear error instead of silently returning garbage.
